actions/submit-signing-request/tests/task.test.ts

```diff
@@ -218,4 +228,47 @@ it('no retries for http code 500', async () => {
     const setFailedStub = sandbox.stub(core, 'setFailed');
     await task.run();
     assert.equal(setFailedStub.calledOnce, true);
+});
+
+it('task waits for artifact being downloaded before completing', async () => {
```

Comment:
There's a test case missing that verifies that the retries will stop after the signing request has a final status (but no unsignedArtifactLink)

Reply:
Good point! Added a test for this.